### PR TITLE
WDAPI-645 add device posture integration api

### PR DESCRIPTION
**Pull request by aplunk**
Add device posture integration API support.

## Description

This pr adds CRUD support for the device posture integration API.

## Has your change been tested?

These changes have been tested using the Cloudflare terraform provider.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly. **(TODO)**
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [x] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs.  **(TODO)**

[1]: https://help.github.com/articles/closing-issues-using-keywords/

### CODE REVIEW

File: device_posture_rule.go [hunk 1]
```diff
@@ -9,6 +9,139 @@ import (
 	"github.com/pkg/errors"
 )
 
+// DevicePostureIntegrationConfig contains authentication information for a device posture integration.
```
Comment (jacobbednarz) [inline device_posture_rule.go:12]:
```suggestion
// DevicePostureIntegrationConfig contains authentication information 
// for a device posture integration.
```

File: device_posture_rule.go [hunk 2]
```diff
@@ -9,6 +9,139 @@ import (
...
+		return DevicePostureIntegration{}, ResultInfo{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return devicePostureIntegrationResponse.Result, devicePostureIntegrationResponse.ResultInfo, nil
+}
+
+// DevicePostureIntegration returns a specific device posture integrations within an account.
+//
+// API reference: TODO
+func (api *API) DevicePostureIntegration(ctx context.Context, accountID, integrationID string) (DevicePostureIntegration, ResultInfo, error) {
```
Comment (jacobbednarz) [inline device_posture_rule.go:89]: do we need `ResultInfo` as a return value?

**Review by jacobbednarz — COMMENTED** [review]
couple of nitpicks but LGTM once the public API changes land

File: device_posture_rule.go (hunk 2 above)
Comment (aplunk) [reply device_posture_rule.go:90]: It's not necessary here.

**Pull request merged**